core/graph.hpp

```diff
@@ -348,11 +348,13 @@ class TGraph {
      */
     void AppendTree(unsigned int node_idx, unsigned int conn_idx, int layer, bool forward, TypeTree tree) {
         int sample = ((unsigned int)rand()) % tree.num_subtrees;
-        unsigned int sample_idx = -1;
-        while (sample >= 0) {
-            sample_idx += 1;
+        unsigned int sample_idx = 0;
+        while (sample >= 0 && sample_idx < tree.children.size()) {
             sample -= tree.children[sample_idx].num_subtrees;
         }
+        if (sample_idx >= tree.children.size()) {
```

Comment:
curious if you hit cases where this went out of bounds; if so, may be a bug with the `num_subtrees` computation

Reply:
yes. I observed a OOB on `tree.childern[sample_idx]`.

Reply:
on second thought. maybe this was just an attempt to fix the loop above.